Content.Server/SyndicateTeleporter/SyndicateTeleporterSystem.cs

```diff
@@ -0,0 +1,156 @@
+using Content.Shared.Interaction.Events;
+using Robust.Shared.Map;
+using Content.Shared.Charges.Components;
+using Content.Shared.Charges.Systems;
+using System.Numerics;
+using Content.Shared.Maps;
+using Content.Shared.Physics;
+using Content.Shared.Body.Components;
+using Content.Shared.Body.Systems;
+using Robust.Shared.Audio.Systems;
+using System.Threading.Tasks;
+using Robust.Shared.Random;
+using Content.Server.Popups;
+using Content.Shared.Popups;
+
+namespace Content.Server.SyndicateTeleporter;
+
+public sealed class SyndicateTeleporterSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly SharedChargesSystem _charges = default!;
+    [Dependency] private readonly TurfSystem _turf = default!;
+    [Dependency] private readonly IMapManager _mapMan = default!;
+    [Dependency] private readonly SharedBodySystem _body = default!;
+    [Dependency] protected readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly PopupSystem _popups = default!;
+
+    private const string TeleportEffectPrototype = "TeleportEffect";
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SyndicateTeleporterComponent, UseInHandEvent>(OnUse);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<SyndicateTeleporterComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            if (comp.InWall == true)
+            {
+                comp.Timer += frameTime;
```

Comment:
this is something cursed

Reply:
> this is something cursed

Well, this is what a nuclear bomb uses and works stably.